tests/DCPS/XTypes/Pub/XTypesPublisher.cpp

```diff
@@ -256,12 +256,27 @@ int ACE_TMAIN(int argc, ACE_TCHAR* argv[])
   DataWriter_var dw = pub->create_datawriter(topic, dw_qos, 0,
     DEFAULT_STATUS_MASK);
 
-  ACE_OS::sleep(ACE_Time_Value(7, 0));
+  DDS::StatusCondition_var condition = expect_to_match ? dw->get_statuscondition() : topic->get_statuscondition();
+  condition->set_enabled_statuses(expect_to_match ? DDS::PUBLICATION_MATCHED_STATUS : DDS::INCONSISTENT_TOPIC_STATUS);
+  DDS::WaitSet_var ws = new DDS::WaitSet;
+  ws->attach_condition(condition);
+
+  DDS::ConditionSeq conditions;
+  DDS::Duration_t timeout = { 10, 0 };
+  if ((ws->wait(conditions, timeout) != DDS::RETCODE_OK) && expect_to_match) {
+    ACE_ERROR((LM_ERROR, "ERROR: %C condition wait failed for type %C\n",
+      expect_to_match ? "PUBLICATION_MATCHED_STATUS" : "INCONSISTENT_TOPIC_STATUS", type.c_str()));
+    failed = 1;
+  }
```

Comment:
```suggestion
  if (ws->wait(conditions, timeout) != DDS::RETCODE_OK) {
    ACE_ERROR((LM_ERROR, "ERROR: %C condition wait failed for type %C\n",
      expect_to_match ? "PUBLICATION_MATCHED_STATUS" : "INCONSISTENT_TOPIC_STATUS", type.c_str()));
    failed = 1;
  }
```
It won't print the log message if `expect_to_match == false` and the `wait` call fails.

Reply:
Good catch - leftover from the original logic. Thanks!